assets/js/frontend/add-to-cart.js

```diff
@@ -76,33 +76,35 @@ jQuery( function( $ ) {
 			});
 
 			// Trigger event.
-			$( document.body ).trigger( 'adding_to_cart', [ $thisbutton, data ] );
-
-			e.data.addToCartHandler.addRequest({
-				type: 'POST',
-				url: wc_add_to_cart_params.wc_ajax_url.toString().replace( '%%endpoint%%', 'add_to_cart' ),
-				data: data,
-				success: function( response ) {
-					if ( ! response ) {
-						return;
-					}
-
-					if ( response.error && response.product_url ) {
-						window.location = response.product_url;
-						return;
-					}
-
-					// Redirect to cart option
-					if ( wc_add_to_cart_params.cart_redirect_after_add === 'yes' ) {
-						window.location = wc_add_to_cart_params.cart_url;
-						return;
-					}
-
-					// Trigger event so themes can refresh other areas.
-					$( document.body ).trigger( 'added_to_cart', [ response.fragments, response.cart_hash, $thisbutton ] );
-				},
-				dataType: 'json'
-			});
+			var valid = $( document.body ).triggerHandler( 'adding_to_cart', [ $thisbutton, data ] );
```

Comment:
This variable's name could probably better represent its purpose.

Reply:
How about `validate_ajax_adding_to_cart` ?

@ObliviousHarmony Would you take a look at what was proposed here: https://github.com/woocommerce/woocommerce-gateway-paypal-express-checkout/issues/694#issuecomment-597408337

Do you prefer that? Does it matter? The end result is about the same.

Reply:
Perhaps `should_send_request`? While your specific use-case may be used for validation, this hook itself doesn't necessarily need to.

I'm more partial to this solution because it feels more in-line with what you'd expect of a WordPress plugin hook. Messing around with the validations API just seems unnecessary in this specific case.

Reply:
I didn't mean to imply we'd need the validations API, just the idea of passing an object property. But I'm fine with this way too... it does seem slightly more familiar.

Reply:
I went with `should_send_ajax_request.adding_to_cart` I don't particularly like it, but `should_send_request` by itself doesn't seem specific enough since there are lots of cases where things are sent to ajax.

Reply:
Sorry @helgatheviking, I meant the `valid` variable, not the event name! `adding_to_cart` was great because it was a clean non-breaking change!